en/developer_guide/routing.rst

```diff
@@ -0,0 +1,92 @@
++==========+========+==================+==========+=================+=============================+
+| 0        | http   | dev.example.com  | fr       | fr              | http://dev.example.com/fr   |
++----------+--------+------------------+----------+-----------------+-----------------------------+
+| 1        | http   | dev.example.com  | en       | none            | http://dev.example.com      |
++----------+--------+------------------+----------+-----------------+-----------------------------+
+| 2        | http   | test.example.com | fr       | fr              | http://test.example.com/fr  |
++----------+--------+------------------+----------+-----------------+-----------------------------+
+| 3        | https  | test.example.com | en       | none            | https://test.example.com    |
++----------+--------+------------------+----------+-----------------+-----------------------------+
+
+
+URL Pattern
+-----------
+
+The url pattern is added to the ``base URL`` to generate the url.
+For node, the url pattern can be choose by contributor or it is generated automatically from the page title.
+
+See ``Url pattern`` chapter from `Node Configuration`_ for more details.
+
+Redirection
+-----------
+
+Open Orchestra allows user to determine redirection in the back office.
+The redirection form contains some arguments:
+
+* **site name**: contextual site which trigger redirection
+* **site language**: contextual language which trigger redirection
+* **pattern to redirect**: pattern that will be the URL of the redirection
+* **node where the redirection is headed**: redirect to a node
+* **Url to redirect to**: redirect to url if node is not specify
+* **permanent redirection**: Use to adapt response code
+
+Implementation
+--------------
+
+``open-orchestra-front-bundle`` contains all routing files in the ``routing`` folder.
+
+
+Route Loader
+~~~~~~~~~~~~
+
+In Open Orchestra, ``Route Loader`` provide route from database collections:
+
+* ``DatabaseRouteLoader`` generate routes for different nodes.
+Routes in loader are linked with ``NodeController`` and take languages, node, site and alias in parameters.
+Route name is generated by two id : ``AliasID``_``NodeId``.
+
+* ``RedirectionLoader`` generate routes for different redirection.
+Routes are linked to the redirect controller from the frameworkBundle.
+Route name is generated by two id : ``AliasID``_``RedirectionID``.
+
+See `Symfony Loader documentation`_ for more information.
+
+URL Generator
+~~~~~~~~~~~~~
+
+``OpenOrchestraUrlGenerator`` generates URL from route.
+The URL Generator is override to take `RequestStack`_ in argument and get the ``aliasId`` from it.
+It will be used to adapt this route name with their generate by Open Orchestra ``Route Loader``.
+
+Generator dumper
+~~~~~~~~~~~~~~~~
+
+``OpenOrchestraGeneratorDumper`` extend ``PhpGeneratorDumper``, it generate cache for route and avoid to request database for each connection.
```

Comment:
it will dump the routing in the Symfony cache file to avoid any request to the database